Mage.Sets/src/mage/cards/g/GaleWaterdeepProdigy.java

```diff
@@ -0,0 +1,187 @@
+package mage.cards.g;
+
+import mage.ApprovingObject;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.ChooseABackgroundAbility;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.effects.ContinuousEffect;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.FilterSpell;
+import mage.filter.common.FilterInstantOrSorcerySpell;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.game.stack.Spell;
+import mage.players.Player;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.target.targetpointer.FixedTarget;
+import mage.watchers.common.CastFromHandWatcher;
+
+import java.util.UUID;
+
+/**
+ * @author Rjayz
+ */
+public final class GaleWaterdeepProdigy extends CardImpl {
+
+    public GaleWaterdeepProdigy(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.WIZARD);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Whenever you cast an instant or sorcery spell from your hand,
+        // you may cast up to one of the other type from your graveyard.
+        // If a spell cast from your graveyard this way would be put into your graveyard, exile it instead.
+        this.addAbility(
+                new GaleWaterdeepProdigyTriggeredAbility(new GaleWaterdeepProdigyEffect(),
+                        new FilterInstantOrSorcerySpell("an instant or sorcery spell from your hand"),
+                        false),
+                new CastFromHandWatcher());
+
+        // Choose a Background
+        this.addAbility(ChooseABackgroundAbility.getInstance());
+    }
+
+    private GaleWaterdeepProdigy(final GaleWaterdeepProdigy card) {
+        super(card);
+    }
+
+    @Override
+    public GaleWaterdeepProdigy copy() {
+        return new GaleWaterdeepProdigy(this);
+    }
+}
+
+class GaleWaterdeepProdigyTriggeredAbility extends SpellCastControllerTriggeredAbility {
+
+    public GaleWaterdeepProdigyTriggeredAbility(Effect effect, FilterSpell filter, boolean optional) {
```

Comment:
if you're making a custom ability then you don't need to pass things in through the constructor, instead just add them directly within the constructor